App/SB/views/ThreadDetail/index.js

```diff
@@ -109,9 +107,10 @@ class ThreadsDetail extends React.PureComponent {
     })
   }
 
-  _onPhotoSelect = () => {
-    return (photoId) => {
-      this.props.viewPhoto(photoId, this.props.threadId)
+  _onPhotoSelect = (photo: Photo) => {
+    return () => {
+      this.props.viewPhoto(photo.id)
+      this.props.navigation.navigate('PhotoViewer')
```

Comment:
One simplifying change I made is to manage state about which thread/photo is being viewed or operated on in redux, and leave navigation up to the UI layer any time that is possible.

Previously, we were dispatching a redux action that would update some state in the reducer and we would also have a saga listening for the action and the saga would trigger a navigation. Super confusing and hard to follow. 

Now the UI dispatches a redux action to update state about which thread/photo we are viewing and then navigates to the appropriate screen. We don't have to pass any data along in the navigation because the screen we are navigating to can get it from redux.